src/main/java/com/example/finance/mapper/BudgetMapper.java

```diff
@@ -0,0 +1,33 @@
+package com.example.finance.mapper;
+
+import com.example.finance.constants.ApplicationConstants;
+import com.example.finance.model.dto.BudgetDto;
+import com.example.finance.model.dto.CategoryDto;
+import com.example.finance.model.entity.BudgetEntity;
+import com.example.finance.model.entity.CategoryEntity;
+import org.mapstruct.IterableMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+
+import java.util.List;
+
+@Mapper(componentModel = ApplicationConstants.SPRING_COMPONENT_MODEL, uses = {UserAccountMapper.class, CategoryMapper.class})
+public interface BudgetMapper {
+
+    @Mapping(target = "userAccountDto", source = "userAccountEntity")
+    @Mapping(target = "categoryDtos", source = "categoryEntity")
+    @Named("toDto")
+    BudgetDto toDto(BudgetEntity budgetEntity);
+
+    @Mapping(target = "userAccountEntity", source = "userAccountDto")
+    @Mapping(target = "categoryEntity", source = "categoryDtos")
+    @Named("toEntity")
+    BudgetEntity toEntity(BudgetDto budgetDto);
+
+    @IterableMapping(qualifiedByName = "toDto")
+    List<BudgetDto> toListDto(List<BudgetEntity> budgetEntities);
```

Comment:
Try to keep only those methods which you will be using in your application. I think at the start toDto and ToEntity will be enough :)